packages/next/client/index.js

```diff
@@ -307,6 +307,10 @@ function renderReactElement(reactEl, domEl) {
         list.getEntries().forEach(onPerfEntry)
       })
       observer.observe({ entryTypes: ['paint'] })
+    } else {
+      window.addEventListener('load', () => {
```

Comment:
If you use the `buffered: true` flag with `PerformanceObserver`, then you shouldn't need to wait until load, nor should you need to use the performance timeline at all (i.e. `performance.getEntriesByType()`).

And since Chrome/Chromium is the only browser to support paint timing, then you maybe as well just use the buffered flag (supported since M77) as shown in this example: https://web.dev/fcp/#measure-fcp-in-javascript

Also, as per our conversation, you'll probably want to wrap this in a try/catch to prevent an error in Safari ([this issue](https://github.com/GoogleChrome/web.dev/issues/2078) tracks updating the web.dev examples to account for the Safari error issue).